pkg/clients/eventstreamsadminv1/eventstreamsadminv1.go

```diff
@@ -0,0 +1,234 @@
+	if in == nil {
+		return nil
+	}
+	o := []arv1.ConfigUpdate{}
+	for _, m := range in {
+		item := arv1.ConfigUpdate{
+			Name:  reference.ToPtrValue(m.Name),
+			Value: reference.ToPtrValue(m.Value),
+			//ResetToDefault: needs to be a pointer to a boolean value, should be true or false??
+		}
+		o = append(o, item)
+	}
+	return o
+}
+
+// GenerateObservation produces TopicObservation object from *arv1.TopicDetail object.
+func GenerateObservation(in *arv1.TopicDetail) (v1alpha1.TopicObservation, error) {
+
+	o := v1alpha1.TopicObservation{
+		ReplicationFactor:  ibmc.Int64Value(in.ReplicationFactor),
+		RetentionMs:        ibmc.Int64Value(in.RetentionMs),
+		CleanupPolicy:      reference.FromPtrValue(in.CleanupPolicy),
+		Configs:            Generatev1alpha1TopicConfigs(in.Configs),
+		ReplicaAssignments: GenerateReplicaAssignments(in.ReplicaAssignments),
+	}
+	return o, nil
+}
+
+// Generatev1alpha1TopicConfigs generates *v1alpha1.TopicConfigs from *arv1.TopicConfigs
+func Generatev1alpha1TopicConfigs(in *arv1.TopicConfigs) *v1alpha1.TopicConfigs {
+	if in == nil {
+		return nil
+	}
+	o := &v1alpha1.TopicConfigs{
+		CleanupPolicy:     reference.FromPtrValue(in.CleanupPolicy),
+		MinInsyncReplicas: reference.FromPtrValue(in.MinInsyncReplicas),
+		RetentionBytes:    reference.FromPtrValue(in.RetentionBytes),
+		RetentionMs:       reference.FromPtrValue(in.RetentionMs),
+		SegmentBytes:      reference.FromPtrValue(in.SegmentBytes),
+		SegmentIndexBytes: reference.FromPtrValue(in.SegmentIndexBytes),
+		SegmentMs:         reference.FromPtrValue(in.SegmentMs),
+	}
+	return o
+}
+
+// GenerateReplicaAssignments generates []v1alpha1.ReplicaAssignment from []arv1.ReplicaAssignment
+func GenerateReplicaAssignments(in []arv1.ReplicaAssignment) []v1alpha1.ReplicaAssignment {
+	if in == nil {
+		return nil
+	}
+	o := []v1alpha1.ReplicaAssignment{}
+	for _, m := range in {
+		item := v1alpha1.ReplicaAssignment{
+			ID:      ibmc.Int64Value(m.ID),
+			Brokers: GenerateBrokers(m.Brokers),
+		}
+		o = append(o, item)
+	}
+	return o
+}
+
+// GenerateBrokers generates *v1alpha1.ReplicaAssignmentBrokers from *arv1.ReplicaAssignmentBrokers
+func GenerateBrokers(in *arv1.ReplicaAssignmentBrokers) *v1alpha1.ReplicaAssignmentBrokers {
+	if in == nil {
+		return nil
+	}
+	o := &v1alpha1.ReplicaAssignmentBrokers{
+		Replicas: in.Replicas,
+	}
+	return o
+}
+
+// IsUpToDate checks whether current state is up-to-date compared to the given set of parameters.
+func IsUpToDate(in *v1alpha1.TopicParameters, observed *arv1.TopicDetail) (bool, error) {
+	desired := in.DeepCopy()
+	actual, err := GenerateTopicParameters(observed)
+	if err != nil {
+		return false, err
+	}
+
+	// it says logger isn't found even when logger is imported??
+	// l.Info(cmp.Diff(desired, actual, cmpopts.IgnoreTypes(&runtimev1alpha1.Reference{}, &runtimev1alpha1.Selector{}, []runtimev1alpha1.Reference{})))
+
+	// what is it being compared to, ie which fields should be ignored??
```

Comment:
The goal of IsUpToDate() is to compare the desired state (Spec) with the current status (Status) to see if I need to re-apply the Spec. In order to compare, we need to reconstruct a Spec from the current Status and compare with the actual Spec so that I will now if something is changed. In an ideal situation, Spec would always be a subset of Status (all fields in Spec are present in Status) so I would be able to compare all fields. But some API is defined so that some fields present in Spec are not present in Status, and so we need to ignore this from the comparison, and that is what IgnoreFields does